cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/AdminParameters.java

```diff
@@ -27,14 +29,11 @@
  * </pre>
  */
 public class AdminParameters extends AbstractParameters {
-  protected Set<AnomalyType> _disableSelfHealingFor;
-  protected Set<AnomalyType> _enableSelfHealingFor;
-  protected Integer _concurrentInterBrokerPartitionMovements;
-  protected Integer _concurrentIntraBrokerPartitionMovements;
-  protected Integer _concurrentLeaderMovements;
   protected Integer _reviewId;
-  protected Set<Integer> _dropRecentlyRemovedBrokers;
-  protected Set<Integer> _dropRecentlyDemotedBrokers;
+  protected Map<String, ?> _configs;
+  protected Optional<DropRecentBrokersParameters> _dropBrokersParameters;
+  protected Optional<UpdateSelfHealingParameters> _updateSelfHealingParameters;
+  protected Optional<ChangeExecutionConcurrencyParameters> _changeExecutionConcurrencyParameters;
```

Comment:
[There are multiple use cases of this pattern] As indicated by the `Java` language architect `Brian Goetz`, we should almost never use `Optional` as a field of something. He also says that:

```
Of course, people will do what they want.
But we did have a clear intention when adding this feature, and it was not to be a general purpose Maybe type, as much as many people would have liked us to do so.
Our intention was to provide a limited mechanism for library method return types where there needed to be a clear way to represent "no result", and using null for such was overwhelmingly likely to cause errors.
```
*Source: https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555#26328555

I believe we can be more conservative about the use of `Optional` -- i.e. avoid using it when it is not absolutely needed. In our case, is using `null` to check for no result cases possible?

Reply:
Sure, I think that holds. I did hesitate between using `Optional` or not, the reason I choose to use it is because I want to explicitly signal the fact that these `Parameters` object can be null.